nss-ice-phantoms/BangazonSite
# Rr products index details good2

# Description
Issue #1 
When viewing any page on the Bangazon site Product Categories appears in the nav bar.  Clicking the link displays a list of all product categories and next to the category name, the number of products in that category.  Below the name, the first three products in the category will be displayed and the names are hyperlinks to the product detail.

Issue #2 
When viewing any page that contains a hyperlink for a product, clicking the link will show the product detail page containing title, description, quantity available, price/unit, and a button labeled Add to Order.  The quantity available should reflect purchases, not the number originally for sale.  

Notes:  
1.  The details page view is complete.  On any page where you want to create a link to product detail (ie...the Order page that lists all of the products in the order) make sure you point it to the Products controller and the Details method.  See Bangazon\Views\Products\Types.cshtml, Lines 12-14 for an example of how to do this.

2.  There is no specific functionality for adjusting the quantity in the ProductsController.  This functionality should occur in the OrdersController.  When a product is added to an order, the quantity column in the DB should be decremented by the quantity ordered, if an order is cancelled then the quantity ordered should be added back.  Assuming the OrdersController contains this functionality, then the quantity in the DB should always reflect "Quantity Available" and thus there is no reason to address that in the products controller.


## Type of change

# Testing Instructions

1. `git fetch --all`
2. `git checkout rr-products-index-details-good2`
3. Run BangazonSite
4. Verify that "Product Categories" appears in the nav bar and that clicking the link displays a list of all product categories.   Next to the category name, the number of products in that category should be displayed in ( ).  Below the name, the first three products in the category will be displayed and the names are hyperlinks to the product detail.  When viewing any page that contains a hyperlink for a product, clicking the link will show the product detail page containing title, description, quantity available, price/unit, and a button labeled Add to Order.  The quantity available should reflect purchases, not the number originally for sale.  

# Checklist:
